test/inductor/test_torchinductor.py

```diff
@@ -1538,9 +1538,11 @@ def forward(self, attention_scores):
                 (torch.randn(8, 12, 512, 512),),
             )
 
-    # For gpu path, there has a accurcy issue,
-    @unittest.skipIf(HAS_CUDA, "only support cpu conv bn test")
     def test_conv_bn_fuse(self):
+        # For gpu path, there is an accuracy issue
+        if self.device == "cuda":
+            raise unittest.SkipTest("only support cpu conv bn test")
```

Comment:
`unittest.SkipTest` is preferred over `unittest.skipIf`?

Reply:
`skipIf` is only for decorators, no? So I used `raise`, which we also use in the rest of the file. I did test that the pushed version works as expected and fixes the problem.

Note that we _cannot_ use the decorator here for the reasons mentioned above. But I'll repeat for clarity. There are two issues that contribute to this:
- the  `CommonTemplate` class is used to generate actual CUDA/CPU tests later (the code is shared)
- `HAS_CUDA` will return True if you have CUDA and triton installed (see the definition of `HAS_CUDA`).

So if you do have CUDA and triton installed on a machine, the template function will be marked as skipped, which will skip both `_cpu` and `_cuda` tests that are generated from it. Instead, you want to be able to run CPU-only tests.

The fix here is to delay checking until we're in one of the CUDA/CPU classes, where device is defined, e.g., for CPU:

```py
    class CpuTests(TestCase):
        common = check_model
        device = "cpu"  # we check this attr in the tests to make sure we're in the right class

    CommonTemplate.install(CpuTests, "cpu")  # tests are copied to CPU here
```

Let me know if this doesn't answer the question.